packages/peregrine/lib/talons/MyAccount/useNewsletterSubscription.js

```diff
@@ -0,0 +1,73 @@
+import { useCallback, useState } from 'react';
+import { useMutation } from '@apollo/react-hooks';
+import { useToasts } from '@magento/peregrine';
+import { useUserContext } from '../../context/user';
+import { useAwaitQuery } from '../../hooks/useAwaitQuery';
+
+export const useNewsletterSubscription = props => {
+    const { subscribeMutation, customerQuery } = props;
+    const [{ currentUser, isSignedIn }, { getUserDetails }] = useUserContext();
+
+    const [subscribe, setSubscribe] = useState(
+        currentUser && currentUser.hasOwnProperty('is_subscribed')
+            ? currentUser.is_subscribed
+            : false
+    );
+    const [isSubmitting, setIsSubmitting] = useState(false);
+
+    const [subscription, { error: subscriptionError }] = useMutation(
+        subscribeMutation,
+        {
+            fetchPolicy: 'no-cache'
+        }
+    );
+    const fetchUserDetails = useAwaitQuery(customerQuery);
+    const [, { addToast }] = useToasts();
+
+    const errors = [];
+    if (subscriptionError) {
+        errors.push(subscriptionError.graphQLErrors[0]);
+    }
+
+    const toggleSubscription = useCallback(() => {
+        setSubscribe(!subscribe);
+    }, [subscribe, setSubscribe]);
+
+    const handleSubmit = useCallback(
+        async ({ is_subscribed }) => {
+            setIsSubmitting(true);
+            try {
+                if (is_subscribed === undefined) {
+                    is_subscribed = subscribe;
+                }
+                await subscription({
+                    variables: { is_subscribed }
+                });
+                addToast({
+                    type: 'info',
+                    message: 'Save user communications successfully!',
+                    timeout: 3000
+                });
```

Comment:
Will check with UX, but we probably need to remove the toast treatment on success.